release/models/nat/openconfig-nat.yang

```diff
@@ -0,0 +1,965 @@
+module openconfig-nat {
+
+  yang-version "1";
+
+  // namespace
+  namespace "http://openconfig.net/yang/nat";
+
+  prefix "oc-nat";
+
+  // import some basic types
+  import ietf-inet-types { prefix inet; }
+  import ietf-yang-types { prefix yang; }
+  import openconfig-interfaces { prefix oc-if; }
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-acl { prefix oc-acl; }
+
+  // meta
+  organization "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module defines configuration and operational state data
+    for Network Address Translation (NAT) functionality.
+
+    The model supports various NAT types including:
+    - Basic NAT44 (Network Address Translation)
+    - NAPT44 (Network Address Port Translation)
+    - Static mappings
+    - Dynamic address pools
+    - Policy-based NAT using ACLs
+    - Active translation table showing current NAT entries";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2025-07-15" {
+    description
+      "Initial revision";
+    reference "0.1.0";
+  }
+
+  // Identity statements
+
+  identity NAT_TYPE {
+    description
+      "Base identity for NAT types";
+  }
+
+  identity BASIC_NAT44 {
+    base NAT_TYPE;
+    description
+      "Basic NAT44 - translates IPv4 addresses only";
+  }
+
+  identity NAPT44 {
+    base NAT_TYPE;
+    description
+      "Network Address Port Translation - translates both
+      IPv4 addresses and port numbers";
+  }
+
+  identity STATIC_NAT {
+    base NAT_TYPE;
+    description
+      "Static NAT - fixed one-to-one address mappings";
+  }
+
+  // Typedef statements
+
+  typedef protocol-type {
+    type enumeration {
+      enum TCP {
+        description "Transmission Control Protocol";
+      }
+      enum UDP {
+        description "User Datagram Protocol";
+      }
+      enum ICMP {
+        description "Internet Control Message Protocol";
+      }
+      enum ALL {
+        description "All protocols";
+      }
+    }
+    description
+      "Protocol types supported for NAT translation";
+  }
+
+  typedef nat-action {
+    type enumeration {
+      enum TRANSLATE {
+        description
+          "Apply NAT translation";
+      }
+      enum BYPASS {
+        description
+          "Bypass NAT translation";
+      }
+    }
+    description
+      "Action to take for matching traffic";
+  }
+
+  typedef log-level {
+    type enumeration {
+      enum NONE {
+        description "No logging";
+      }
+      enum ERROR {
+        description "Log only errors";
+      }
+      enum INFO {
+        description "Log informational messages";
+      }
+      enum DEBUG {
+        description "Log debug messages";
+      }
+    }
+    description
+      "Logging level for NAT operations";
+  }
+
+  typedef interface-type {
```

Comment:
As discussed in the OC weekly meeting -- the terms "inside" and "outside" are ambiguous and may cause confusion.

Based on our discussion, the model seems to be focused on performing source NAT on traffic moving from a private to a public network. However, NAT may also be performed in the opposite direction (public -> private) or between private/private or public/public networks. 

To reduce iterations in the future, can we update the model to be general/flexible enough to support those other cases in the future, without having to actively support them today?

Reply:
can we also avoid the terms "inside" and "outside"?

Reply:
Do you prefer internal, external for the `interface-type?`
```c
  typedef interface-type {
    type enumeration {
      enum INSIDE {
        description
          "Source interface for NAT address translation.";
      }
      enum OUTSIDE {
        description
          "Destination interface for translated addresses.";
      }
    }
    description
      "NAT interface role: INSIDE or OUTSIDE.";
  }
```

Reply:
Discussed in OC operator meeting, existing terms are OK